### PR TITLE
Dynamic project and yearbook pages

### PR DESCRIPTION
**Pull request by craig-riecke**
- Removed Register button to make room for Projects menu item
- Added Projects page, populated by data in meta repo
- Added Help Wanted page
- Added 2017 Yearbook projects, populated dynamically

### CODE REVIEW

File: _includes/_global_js.html [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+<script src="/assets/js/bootstrap.min.js"></script>
+{% for js_name in page.extra_js %}<script src="/assets/js/{{ js_name }}"></script>{% endfor %}
+
+<script>
+  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
+  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
+  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
+  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
+
+  ga('create', 'UA-77346638-1', 'codefordc.org');
```
Comment (seanmarcia) [inline _includes/_global_js.html]: Probably don't want to put in code for dc's google analytics ;)

**Comment by craig-riecke** [comment]
I tried to fork this repo, but could only get it to host as https://craig-riecke.github.io/rubyforgood.html, which made all the assets and links not work.  So you'll need to set up a local host to see what it actually looks like.

File: _includes/_global_js.html (hunk 1 above)
Comment (craig-riecke) [reply _includes/_global_js.html]: Good point!

File: _includes/_global_js.html [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            this.href == window.location.href + "index.html" ) {
+
+            $(this).parent().addClass("active");
+
+          }
+        });
+    });
+</script>
+
+{% if page.deeplink == true %}
```
Comment (csexton) [inline _includes/_global_js.html]: I think we can remove this entire section, no pages currently set `deeplink: true`.

File: _includes/_global_js.html [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      $("#navbar-nav a").each(function() {
+        if (this.href == window.location.href ||
+            this.href + "index.html" == window.location.href ||
+            this.href == window.location.href + "index.html" ) {
+
+            $(this).parent().addClass("active");
+
+          }
+        });
+    });
```
Comment (csexton) [inline _includes/_global_js.html]: Local testing it doesn't look like this works/is used. I'd prefer to just delete this snippet.

File: _includes/_global_js.html [hunk 4]
```diff
@@ -0,0 +1,55 @@
+<!-- This should be inserted in the page *before* any page-specific JS -->
+<script src="/assets/js/jquery.js"></script>
+<script src="/assets/js/bootstrap.min.js"></script>
+{% for js_name in page.extra_js %}<script src="/assets/js/{{ js_name }}"></script>{% endfor %}
```
Comment (csexton) [inline _includes/_global_js.html]: Any reason we shouldn't just include the `extra_js` files from the project page in the `base.html` layout?

File: _includes/_global_js.html [hunk 5]
```diff
@@ -0,0 +1,55 @@
+<!-- This should be inserted in the page *before* any page-specific JS -->
+<script src="/assets/js/jquery.js"></script>
+<script src="/assets/js/bootstrap.min.js"></script>
```
Comment (csexton) [inline _includes/_global_js.html]: Duplicate includes, both bootstrap and jquery are already included in the `base.html` layout.

File: _includes/_global_js.html (hunk 1 above)
Comment (csexton) [reply _includes/_global_js.html]: I'd just drop the google analytics all together.

File: _includes/projects-yearbook.html [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  </div>
+</div>
+
+<div id="project-list">
+  <!-- populated by js -->
+</div>
+
+<div class="clearfix"></div>
+
+{% include _global_js.html %}
```
Comment (csexton) [inline _includes/projects-yearbook.html]: Looks like there are duplicate includes in this file.

File: _layouts/base.html [hunk 7]
```diff
@@ -24,6 +24,7 @@
   <link rel="stylesheet" href="assets/css/jquery.fancybox.css" type="text/css" media="all" />
   <link rel="stylesheet" href="assets/css/style-green.css" id="mainstyle" type="text/css" media="all" />
   <link rel="stylesheet" href="assets/css/custom.css" id="custom" type="text/css" media="all" />
+  {% for css_name in page.extra_css %}<link rel="stylesheet" href="/assets/css/{{ css_name }}">{% endfor %}
```
Comment (csexton) [inline _layouts/base.html:27]: Just like the JS: No need to special case the CSS includes, we should just load them all in `base.html`

File: helpwanted.html [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        box.find(".comments").removeClass("hidden").text(issue.comments);
+                box.find(".project").removeClass("hidden").text(issue.project);
+        $.each(issue.labels, function(i, label){
+          box.find(".labels").removeClass("hidden").append("<span class='btn btn-sm btn-default'>"+label.name+"</span>");
+        });
+      });
+    }
+  });
+};
+</script>
```
Comment (csexton) [inline helpwanted.html:69]: I love this page! Nice work :heart:

**Review by csexton — CHANGES_REQUESTED** [review]
Ran things locally and it all looked great. This is really awesome I just have a few nit-picky things I'd like to get cleaned up. 

The projects, yearbook and help wanted pages are absolutely wonderful.

**Comment by craig-riecke** [comment]
All righty, I took all of your suggestions.  It turns out the _global.js stuff was funky due to page loading timing issues.  Once I fixed those, _global.js wasn't needed anymore so I removed it.   Lemme know if anything is still amiss.

**Review by csexton — APPROVED** [review]
Thanks for the updates @craig-riecke, much simpler. I ran everything locally and it all looked great -- even the new slack section on the project page.

:shipit:

**Pull request merged**